cmd/rendmp/src/lib.rs

```diff
@@ -1002,55 +1047,959 @@ fn rendmp_open_pin(
+                if let Err(e) = next()? {
+                    bail!("failed to set VMON_ON for {rail}: {e}",);
+                }
+                if let Err(e) = next()? {
+                    bail!("failed to set VMON_OFF for {rail}: {e}",);
+                }
+            }
+            SupportedDevice::RAA229618 => {
+                if let Err(e) = next()? {
+                    bail!("failed to set VIN_ON for {rail}: {e}",);
+                }
+                if let Err(e) = next()? {
+                    bail!("failed to set VIN_OFF for {rail}: {e}",);
+                }
+
+                // Clear bit 0 of DMA register EA5B and write it back
+                // (the name part_fast_add comes from Power Navigator)
+                let mut part_fast_add = match next()? {
+                    Ok(v) => v.expect_read_dma()?,
+                    Err(e) => {
+                        bail!("worker.failed to read EA5B for rail {rail}: {e}")
+                    }
+                };
+                part_fast_add &= !1; // clear bit 0
+                let reg = (0xEA5B + rail * 0x80) as u16;
+                worker.write_dma(addr, reg, part_fast_add)?;
+            }
+        }
+
+        let mut disable_fault_value = match next()? {
+            Ok(v) => v.expect_read_dma()?,
+            Err(e) => bail!("failed to read {disable_fault_reg:x}: {e}"),
+        };
+        // clear bits 0:3 and 17:18;
+        // this disables all input and output voltage faults.
+        disable_fault_value &= !(0b1111 | (0b11 << 17));
+        worker.write_dma(
+            addr,
+            disable_fault_reg + rail as u16,
+            disable_fault_value,
+        )?;
+    }
+
+    let vout_mode = match next()? {
+        Ok(v) => v.expect_read_byte()?,
+        Err(e) => bail!("failed to read VOUT_MODE: {e}"),
+    };
+    let vout_mode = || pmbus::commands::VOUT_MODE::CommandData(vout_mode);
+
+    match next()? {
+        Ok(v) => v.expect_write_dma()?,
+        Err(e) => {
+            bail!("failed to disable all phases: {e}")
+        }
+    }
+
+    if let Some(i) = iter.next() {
+        bail!("expected iterator to be done; got {i:?} instead");
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Check the results of the modify-write portion of register changes
+    let results = worker.run(core)?;
+    let mut iter = results.iter();
+    let mut next = || iter.next().ok_or_else(|| anyhow!("early termination"));
+    for (rail, ..) in rail_indexes.iter().enumerate() {
+        match next()? {
+            Ok(v) => v.expect_write_word32()?,
+            Err(e) => {
+                bail!("failed to modify LOOPCFG for rail {rail}: {e}")
+            }
+        }
+        match next()? {
+            Ok(v) => v.expect_write_word()?,
+            Err(e) => {
+                bail!("failed to modify PEAK_OCUC_COUNT for rail {rail}: {e}")
+            }
+        }
+        if dev == SupportedDevice::RAA229618 {
```

Comment:
Nit - would it be worth using a `match` here instead so if we add to `SupporteDevice` in the future, we get a compile-time reminder to check whether the new device needs anything here?

Reply:
Sure, although this decoding is paired with a write operation in the loop above, so it's not strictly necessary.  Done in eb4676c7ad703b0c2769543fe837c1d5e6662794